stats/src/test/java/io/airlift/stats/TestQuantileDigest.java

```diff
@@ -815,6 +816,187 @@ public void testSerializationRandom()
+        assertEquals(iterator.count(), 1D, 0.0001);
+        assertEquals(iterator.cumulativeCount(), 3D, 0.0001);
+        assertEquals(iterator.lowerBoundValue(), 5L);
+        assertEquals(iterator.upperBoundValue(), 5L);
+
+        assertTrue(iterator.next());
+        assertEquals(iterator.lowerBoundQuantile(), 0.733333333D, 0.0001);
+        assertEquals(iterator.upperBoundQuantile(), 0.8D, 0.0001);
+        assertEquals(iterator.count(), 1D, 0.0001);
+        assertEquals(iterator.cumulativeCount(), 4D, 0.0001);
+        assertEquals(iterator.lowerBoundValue(), 4L);
+        assertEquals(iterator.upperBoundValue(), 4L);
+
+        assertTrue(iterator.next());
+        assertEquals(iterator.lowerBoundQuantile(), 0.333333333D, 0.0001);
+        assertEquals(iterator.upperBoundQuantile(), 0.733333333D, 0.0001);
+        assertEquals(iterator.count(), 6D, 0.0001);
+        assertEquals(iterator.cumulativeCount(), 10D, 0.0001);
+        assertEquals(iterator.lowerBoundValue(), 3L);
+        assertEquals(iterator.upperBoundValue(), 3L);
+
+        assertTrue(iterator.next());
+        assertEquals(iterator.lowerBoundQuantile(), 0.066666667D, 0.0001);
+        assertEquals(iterator.upperBoundQuantile(), 0.333333333D, 0.0001);
+        assertEquals(iterator.count(), 4D, 0.0001);
+        assertEquals(iterator.cumulativeCount(), 14D, 0.0001);
+        assertEquals(iterator.lowerBoundValue(), 2L);
+        assertEquals(iterator.upperBoundValue(), 2L);
+
+        assertTrue(iterator.next());
+        assertEquals(iterator.lowerBoundQuantile(), 0D, 0.0001);
+        assertEquals(iterator.upperBoundQuantile(), 0.066666667D, 0.0001);
+        assertEquals(iterator.count(), 1D, 0.0001);
+        assertEquals(iterator.cumulativeCount(), 15D, 0.0001);
+        assertEquals(iterator.lowerBoundValue(), 0L);
+        assertEquals(iterator.upperBoundValue(), 0L);
+
+        assertFalse(iterator.next());
+    }
+
+    @Test
+    public void testIteratorSingleElement()
+    {
+        QuantileDigest digest = new QuantileDigest(0.1);
+        addAll(digest, asList(10));
+        QuantileDigest.QuantileDigestIterator iterator = digest.iterator();
+
+        assertTrue(iterator.next());
+        assertEquals(iterator.upperBoundValue(), 10);
+        assertEquals(iterator.lowerBoundValue(), 10);
+        assertEquals(iterator.count(), 1.0);
+        assertEquals(iterator.cumulativeCount(), 1.0);
+        assertEquals(iterator.lowerBoundQuantile(), 0.0);
+        assertEquals(iterator.upperBoundQuantile(), 1.0);
+        assertFalse(iterator.next());
+    }
+
+    @Test
+    public void testNoElement()
+    {
+        QuantileDigest digest = new QuantileDigest(0.1);
+        QuantileDigest.QuantileDigestIterator iterator = digest.iterator();
+        assertFalse(iterator.next(), "Expected no calls to next() to return true");
+    }
+
+    @Test(expectedExceptions = NoSuchElementException.class)
```

Comment:
This is dangerous because we don't know which method threw. It might have been one of the calls we expected to succeed. When you have a complex method, it's better to assert the throws directly:
```java
assertThatThrownBy(() -> iterator.upperBoundQuantile()).isInstanceOf(NoSuchElementException.class)